### ISSUE TITLE
Very slow boot when storage folder contains subfolders

### ISSUE THREAD

**Issue by RomainVandevoorde** [classic-o/nova-media-library#42]
**Describe the bug**
NovaMediaLibrary massively slows down all Nova requests when the configured folder contains subfolders.

**To Reproduce**
Have multiple levels of folders (in my case, four levels) in the configured disk and all Nova requests are slowed down. The more subfolders you have, the slower it gets.
After some testing, I have found the problematic function is Helper::directories(), called by NovaMediaLibrary::config() during Nova's boot process.

**Screenshots**
Example: a Nova script request.
With NovaMediaLibrary tool enabled:
![tool-activated](https://user-images.githubusercontent.com/38908146/94024642-f6d1ab80-fdb7-11ea-8d56-b7e3689f10f5.PNG)
With the tool removed from NovaServiceProvider::tools():
![tool-deactivated](https://user-images.githubusercontent.com/38908146/94024720-08b34e80-fdb8-11ea-8f88-005a1e05010b.PNG)
With the tool enabled and some subfolders removed from the storage:
![less-subfolders](https://user-images.githubusercontent.com/38908146/94024786-18329780-fdb8-11ea-925c-c51058958f8e.PNG)

**Additional context**
Strangely enough, when I test the function (Helper::directories()) in Tinker in the exact same environment, the operation isn't slow at all. It might be a Laravel bug, I honestly have no idea.
Anyway, I think it's a bad practice to read the filesystem during the tool's boot process. It's a slow operation and the tool is booted at every Nova request. I suggest moving the Helper::directories() operation to an API endpoint called by the Tool component.

**Comment by classic-o** [comment]
Hi. Checked it. My execution time **0.075105905532837 s** with 6 levels folders
Check with this code after parse folders: **dd(microtime(true) - LARAVEL_START);**

How many files do you have?

**Comment by RomainVandevoorde** [comment]
Hey ! Yeah sorry I wasn't specific enough, the folders are not empty. There is a total of 3121 files in there.

I pasted your piece of code at NovaMediaLibrary.php line 51 and the result is pretty much the same: 2.3618609905243
This
```php
        $start = microtime(true);

        if ( 'folders' == $config['store'])
                $config['folders'] = Helper::directories();

        dd(microtime(true) - $start);
```
Gives the following result: **2.1646671295166**

**Comment by classic-o** [comment]
fixed in 1.0.4

**Comment by RomainVandevoorde** [comment]
It's working great after the update !

Thank you for your quick reaction and for this great package ! 💯
